frontend/src/metabase/dashboard/components/add-card-sidebar/QuestionPicker.jsx

```diff
@@ -34,9 +34,9 @@ QuestionPicker.propTypes = {
 
 function QuestionPicker({ onSelect, collectionsById, getCollectionIcon }) {
   const dashboard = useSelector(getDashboard);
-  const initialCollection = dashboard.id;
+  const dashboardCollection = dashboard.collection ?? ROOT_COLLECTION;
```

Comment:
We have some discrepancies between some APIs. For the root collection some API returns the root collection and the id of `root`, [but for dashboard APIs](https://github.com/metabase/metabase/pull/34827#issuecomment-1772602403), it returns neither `collection` nor `collection_id` at all.

So we need to ensure we get the root collection when those values are null.

Note: Please don't mind `const initialCollection = dashboard.id;`, it's a typo from https://github.com/metabase/metabase/pull/35135, and I don't want to rerun CI for all the downstream PRs.